src/main/java/ux/MainWindow.java

```diff
@@ -0,0 +1,60 @@
+//@@author 9fc70c892-reused
+//Source: https://se-education.org/guides/tutorials/javaFx.html
+//Attempted to write my own but took too much time.
+//@@author
+package ux;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private core.Duke duke;
+
+    private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
```

Comment:
type here? vvalueProperty might be valueProperty

Reply:
Hi thank you for pointing this out.

I [have checked](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/ScrollPane.html#vvalueProperty--). And [it does not seem to be a mistake](https://se-education.org/guides/tutorials/javaFxPart4.html#using-controllers), but I will double check if it affects the code flow.